src/plugins/intel_npu/tools/single-image-test/main.cpp

```diff
@@ -2084,6 +2114,12 @@ static int runSingleImageTest() {
             const FilesForModelInputs &inputFiles = inputFilesForOneInfer[numberOfTestCase];
             OPENVINO_ASSERT(inputFiles.size() == inputsInfo.size(), "Number of input files ", inputFiles.size(),
                             " doesn't match network configuration ", inputsInfo.size());
+            const RefFilesForModelOutputs &refFiles = refFilesForOneInfer.empty() ? RefFilesForModelOutputs{}
+                                                                                 : refFilesForOneInfer[numberOfTestCase];
+            if (!FLAGS_ref_results.empty()) {
+                OPENVINO_ASSERT(refFiles.size() == outputsInfo.size(), "Number of reference files ", refFiles.size(),
+                " doesn't match network configuration ", outputsInfo.size());
```

Comment:
+100500 for printing troubleshooting info's!
I'd only suggest emphasize that it must match **output count** by amending this `doesn't match network output count: ', outputsInfo.size()`